src/internal/cmds.ts

```diff
@@ -22,3 +25,21 @@ export function setState(view: IObjectRef<VegaView>, state: any) {
 export function stateCompressor(path: ActionNode[]) {
   return lastOnly(path, CMD_SET_STATE, (n) => n.requires[0].hash);
 }
+
+export async function setSpecImpl(inputs: IObjectRef<any>[], parameter: any) {
+  const view = await inputs[0].v;
+  const old = view.setSpecImpl(parameter.spec);
+  return {
+    inverse: setSpec(inputs[0], old)
+  };
+}
+
+export function setSpec(view: IObjectRef<App>, spec: Spec, title: string = 'Vega Example') {
+  return action(meta(`Open '${title}' Example`, cat.data, op.update), CMD_SET_SPEC, setSpecImpl, [view], {
+    spec
+  });
+}
+
+export function specCompressor(path: ActionNode[]) {
+  return lastOnly(path, CMD_SET_SPEC, (n) => n.requires[0].hash);
```

Comment:
actually since the two actions are dependent on each other you need a more sophisticated compressor.